### ISSUE TITLE
brew install fail: C compiler cannot create executables

### ISSUE THREAD

**Issue by sahrens** [facebook/watchman#114]
I was getting the  "ERROR  Watcher took too long to load" issue in React Native, and the troubleshooting said to uninstall and re-install watchman --head.  First thing I noticed was watchman wasn't installed via brew - maybe an fb chef issue?

```
06/24/2015 10:18:19 sahrens-mbp : ~/src/fbobjc-hg/Libraries/FBReactKit
% watchman version
{
    "version": "3.2.0"
}
06/24/2015 10:19:01 sahrens-mbp : ~/src/fbobjc-hg/Libraries/FBReactKit
% brew uninstall watchman
Error: No such keg: /usr/local/Cellar/watchman
```

Then I tried installing anyway:

```
06/24/2015 10:20:17 sahrens-mbp : ~/src/fbobjc-hg/Libraries/FBReactKit
% brew install --HEAD watchman
==> Installing dependencies for watchman: autoconf, automake, pkg-config, pcre
...
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in `/private/tmp/watchman20150624-8275-1epq85s':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

And I saw this in config.log:

```
ld: building for iOS Simulator, but linking against dylib built for MacOSX file '/usr/lib/libSystem.dylib' for architecture x86_64
```

(full config.log: https://gist.github.com/sahrens/162598cd4070965eb604)

Tried googling, installed latest Xcode Command Line Tools, tried setting HOMEBREW_CC=gcc-5 (broke in a different way), no dice.

```
% gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.2.sdk/usr/include/c++/4.2.1
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
```

brew doctor complains about some static libs and dylibs, maybe from chef? https://gist.github.com/61f6bb44e839c7fdddc4

**Comment by sahrens** [comment]
I have Xcode 6.3.2

**Comment by wez** [comment]
The build problems are almost certainly a local to your machine issue; we haven't changed anything with the watchman build machinery in some time, and I just tried rebuilding and installing via homebrew just now; it worked fine.  Perhaps worth noting, although I doubt it is significant, I had an update to the xcode cli tools that got installed last night.

Separately from that, both brew and the FB deployment system are shipping 3.2.  We tagged 3.3 this week but there have been no changes to the service since then that might impact the "took too long to load" error.

However, we did release fb-watchman 1.0.0 to npm yesterday that switches the nodejs client for watchman from using JSON to use our [BSER](https://facebook.github.io/watchman/docs/bser.html) encoding which resolves a dramatic quadratic-wrt-size-of-response perf problem in a JSON stream module that we were using.  Perhaps that will quash that class of error? https://github.com/facebook/watchman/issues/113 has more on that.

My suggestion is to either let chef keep you on 3.2 or just do a `brew install watchman` and let brew download the bottled watchman binary.

I can't help you resolve your local build issue I'm afraid; I'm just not intimate enough with the FB specifics to give you helpful advice.
